daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
+	priorDrivers := scanPriorDrivers(root)
+	for _, name := range priority {
+		if name == "vfs" {
+			// don't use vfs even if there is state present.
+			continue
+		}
+		for _, prior := range priorDrivers {
+			// of the state found from prior drivers, check in order of our priority
+			// which we would prefer
+			if prior == name {
+				driver, err = GetDriver(name, root, options)
+				if err != nil {
+					if err == ErrNotSupported || err == ErrPrerequisites || err == ErrIncompatibleFS {
```

Comment:
(Hmf GitHub cache (only noticed the comments above))

Yes, I think if there's *any* doubt; ask the user.

Users searching for "my images disappeared" problems cluttering the issue tracker (or worse; blogging about it first) isn't the best user experience.

A clear error (if possible) with clear instructions seems the better choice.